### PR TITLE
[0256] Refactor Fuel Information Related Vehicle Data

### LINKED ISSUE #1223: [SDL 0256] Refactor Fuel Information Related Vehicle Data
Proposal: [Refactor Fuel Information Related Vehicle Data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0256-Refactor-Fuel-Information-Related-Vehicle-Data.md)

> This proposal is to consolidate Fuel Information related vehicle data so that it supports multiple types of propulsion sources used by the vehicle.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/854

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with revisions.  The author will make the following changes to the proposal: add `KILOGRAM` to `CapacityUnit`; change maxValue of `level` and `capacity` in `FuelRange` to larger upper limit (1000000); fix `fuelLevel` and `fuelLevel_State` typos; update descriptions of `fuelLevel` and `fuelLevel_State` to state the param is deprecated as of x.x.x of the rpc spec, please see `fuelRange`; modify description of `fuelRange` param to be more descriptive of its extended capabilities.  The author can work with the Project Maintainer to finalize this description.

> The proposal .md file was updated to reflect these revisions on 11/14/19.

### PR DESCRIPTION
**Pull request by kostyaBoss**
Fixes #1223 

This PR is **[ready]** for review.

### Risk
This PR makes **[minor]** API changes.

### Testing Plan
Covered by existing and added unit tests

#### Unit Tests
New unit tests covering introduced structures were added

### Summary
Applied [0256 Refactor Fuel Information Related Vehicle Data](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0256-Refactor-Fuel-Information-Related-Vehicle-Data.md) changes

#### Enhancements
   - Add CapacityUnit
   - Refactor Fuel information
   - Fix tests

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by kostyaBoss** [comment]
@mrapitis , @atiwari9 please review the changes

**Comment by santhanamk** [comment]
Hi @kostyaBoss I reviewed your PR.  I suggest you make the below changes.

**Source code changes:**

_VehicleDataType.java:_
 - Update VEHICLEDATA_FUELLEVEL to show it goes until version X.x
 - Update VEHICLEDATA_FUELLEVEL_STATE to show it goes until version X.x

_SubscribeVehicleData.java:_
 - can add annotation "@see FuelRange struct for more details" for the fuelRange setter/getter

_UnsubscribeVehicleData:_
 - @Deprecated tag needed on setFuelLevel and getFuelLevel
 - can add annotation "@see FuelRange struct for more details" for the fuelRange setter/getter

_GetVehicleData:_
 - can add annotation "@see FuelRange struct for more details" for the fuelRange setter/getter

_SubscribeVehicleDataResponse:_
 - @Deprecated tag needed on setFuelLevel and getFuelLevel 
 - can add annotation "@see FuelRange struct for more details" for the fuelRange setter/getter

_UnsubscribeVehicleDataResponse:_
 - @Deprecated tag needed on setFuelLevel and getFuelLevel
 - can add annotation "@see FuelRange struct for more details" for the fuelRange setter/getter

_GetVehicleDataResponse:_
 - @Deprecated tag needed on setFuelLevel and getFuelLevel
 - can add annotation "@see struct FuelRange for details."

_OnVehicleData:_
 - @Deprecated tag needed on setFuelLevel and getFuelLevel
 - can add annotation "@see struct FuelRange for details."

**Unit Test changes:**

_Validator.java:_
 - validateFuelRange function needs to check 4 new parameters in the for loop for item1 and item2.

_OnVehicleDataTests.java:_
 - fuelRangeObj.put needs to be done for the 4 new parameters

_GetVehicleDataResponseTests.java:_
 - fuelRangeObj.put needs to be done for the 4 new parameters

_onVehicleDataTests.java:_
 - fuelRangeObj.put needs to be done for the 4 new parameters

_VehicleDataHelper.java:_
 - FuelRange's 4 new parameters need 4 new variables following FUEL_RANGE_TYPE, FUEL_RANGE_RANGE
 - FUEL_RANGE generic object needs setters for 4 new parameters

**Comment by kostyaBoss** [comment]
@santhanamk updated according to provided suggestions, please review

**Comment by santhanamk** [comment]
@kostyaBoss I reviewed your changes.  They look good.  I have a few more suggestions:

**Source code changes:**

_VehicleDataType.java:_
 - change until Smart Device Link X.x to until Smart Device Link 6.0

**Unit Test changes:**

_Validator.java:_
 - in the function `validateFuelRange(...)` change `if (item1.get(i).getRange() != item2.get(i).getRange()) {`
      `return false;`
            `}`
	to
   `if (!item1.get(i).getRange().equals(item2.get(i).getRange())) {`
      `return false;`
            `}`

**Comment by kostyaBoss** [comment]
@santhanamk Thanks for your feedback, I've updated PR, please check it

**Comment by santhanamk** [comment]
@kostyaBoss I checked the code and it looks good.  Thanks.

**Comment by kostyaBoss** [comment]
@joeygrover , @theresalech Hi! Please review changes in this PR

**Comment by theresalech** [comment]
@kostyaBoss, can you please confirm that Ford has reviewed and approved this PR?  I do not see any completed reviews yet.  Thanks!

**Comment by kostyaBoss** [comment]
@theresalech The @santhanamk has approved this PR

**Comment by santhanamk** [comment]
@theresalech this PR is ready for Livio review.

**Comment by kostyaBoss** [comment]
@JulianKast Hi! I've made rollback of formatting changes and fixed method order to make diff clear. Please, check the changes

**Comment by JulianKast** [comment]
@kostyaBoss HI! Same thing as with the [0253] New vehicle data StabilityControlsStatus, I still see formatting changes in most of the files. For example GetVehicleData. Can we remove all the formatting changes from all modified files in this PR?

**Comment by kostyaBoss** [comment]
@JulianKast Hi! It seems there were some additional changes I didn't mention from my IDE's side. I've made the rollback of changes manually right now, please, check the updated version. Let me know if there is something else I should update

**Comment by JulianKast** [comment]
@kostyaBoss HI, A lot of the problems are the same on the New Vehicle Data PR's. Let's focus on https://github.com/smartdevicelink/sdl_java_suite/pull/1395 then we will revisit this one when that one is complete.

**Comment by kostyaBoss** [comment]
@JulianKast Hi! Regarding PR #1395, the @bilal-alsharifi has approved it. The problem with "unclear difs" comes from different line [comment](https://github.com/smartdevicelink/sdl_java_suite/pull/1395#issuecomment-669982922). I and @vladmu noticed that issue comes from the wrong line endings, files are containing different types of endings and we can't fix it. There were pushed commits to reduce the number of changes and make difs as clear as possible. There were also commits to align this PR with latest version of the dev branch

cc @bilal-alsharifi , @vladmu

**Comment by kostyaBoss** [comment]
@JulianKast, @RHenigan let's focus on this PR

**Comment by kostyaBoss** [comment]
@JulianKast , @RHenigan check the PR please

**Comment by kostyaBoss** [comment]
@RHenigan , @JulianKast regarding failed CI issue - I don't really know the reason, but have found in logs this exception 
```

com.smartdevicelink.test.streaming.video.SdlRemoteDisplayTest > testTouchEvents[test(AVD) - 10] FAILED 
	java.lang.SecurityException: Neither user 10103 nor current process has android.permission.WAKE_LOCK.
	at android.os.Parcel.createException(Parcel.java:2071)
```

I've tried to reproduce this issue locally, but was unable to get this. Could you please restart the CI?

File: base/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java [hunk 1]
```diff
@@ -153,17 +153,23 @@ public VehicleDataResult getRpm() {
         return (VehicleDataResult) getObject(VehicleDataResult.class, KEY_RPM);
     }
     /**
-     * Sets Fuel Level
-     * @param fuelLevel a VehicleDataResult related to Fuel Level
+     * Sets the fuelLevel.
```
Comment (JulianKast) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java:156]: Edit KEY_FUEL_LEVEL and KEY_FUEL_LEVEL_STATE above like:

```
    /**
     * @deprecated
     */
    @Deprecated
    public static final String KEY_FUEL_LEVEL = "fuelLevel";
    /**
     * @deprecated
     */
    @Deprecated
    public static final String KEY_FUEL_LEVEL_STATE = "fuelLevel_State";
```

File: base/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java [hunk 2]
```diff
@@ -153,17 +153,23 @@ public VehicleDataResult getRpm() {
...
     /**
-     * Gets Fuel Level
-     * @return a VehicleDataResult related to FuelLevel
+     * Gets the fuelLevel.
+     *
+     * @return VehicleDataResult The fuel level in the tank (percentage). This parameter is deprecated starting RPC Spec
+     * 7.0, please see fuelRange.
      */
     @SuppressWarnings("unchecked")
+    @Deprecated
```
Comment (JulianKast) [inline base/src/main/java/com/smartdevicelink/proxy/rpc/UnsubscribeVehicleDataResponse.java:172]: Deprecate getter and setter for fuleLevelState below as well

**Comment by theresalech** [comment]
Hi @kostyaBoss, can you please advise when this PR will be ready for re-review? As we are nearing the end of our development timeline for this release, we want to make sure all PRs are addressed as soon as possible. Please let us know if you have any questions about our latest review. Thank you!

**Comment by kostyaBoss** [comment]
@theresalech just finished updates, @RHenigan please, check the PR

**Pull request merged**